lib/assessment/code_feature_extractor.py

```diff
@@ -335,19 +360,23 @@ def extract_property_assignment(self, node):
         if property["start"] >= draw_loop_start and property["end"] <= draw_loop_end:
           property["draw_loop"] = True
         new_properties.append(property)
+        self.save_evidence_string(property["start"], property["end"], "property assignment")
```

Comment:
I'm wondering if the evidence strings should be saved in the decision tree instead of during traversal of the parse nodes. Not all of the extracted properties are relevant to the learning goal, but now that you've added line numbers to the movement counters too, we should be able to get all the evidence at once in the decision tree. I think that might also help with future work to have all the evidence and decisions in one place to update. Lmk what you think and if you want to pair on it/ talk about it more